### PR TITLE
Creating a FastAPI application for your comment filtering model.

### PR DESCRIPTION
**Pull request by alireza-abdii**
### **PR Type**
Enhancement

___

### **Description**
- Create FastAPI application for comment toxicity analysis

- Add model training script for Persian text classification

- Include dependencies for ML and web framework

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["Training Data"] --> B["train.py"]
  B --> C["Fine-tuned Model"]
  C --> D["api.py"]
  D --> E["FastAPI Endpoint"]
  F["requirements.txt"] --> B
  F --> D
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>api.py</strong><dd><code>FastAPI application for toxicity detection</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

api.py

<ul><li>Create FastAPI application with <code>/analyze</code> endpoint<br> <li> Load pre-trained Persian BERT model for toxicity detection<br> <li> Implement text normalization and classification pipeline<br> <li> Add error handling for model loading</ul>

</details>

  </td>
  <td><a href="https://github.com/alireza-abdii/AI-Comment-Filter/pull/1/files#diff-1ed02c3495a2cbdfb7310b80b4a533d180f77b750ebf1e23a86d0d93e5872178">+73/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>train.py</strong><dd><code>Model training script for Persian text</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

train.py

<ul><li>Download and process PHICAD Persian dataset<br> <li> Fine-tune BERT model for binary classification<br> <li> Implement training pipeline with evaluation metrics<br> <li> Save trained model and tokenizer</ul>

</details>

  </td>
  <td><a href="https://github.com/alireza-abdii/AI-Comment-Filter/pull/1/files#diff-ed183d67207df065a11e1289f19d34cc2abbc5448dea952683cfe9728c342b95">+159/-0</a>&nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Dependencies</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>requirements.txt</strong><dd><code>Dependencies for ML and API components</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

requirements.txt

<ul><li>Add PyTorch and transformers dependencies<br> <li> Include FastAPI and uvicorn for web service<br> <li> Add Persian text processing library (hazm)<br> <li> Specify ML libraries for training pipeline</ul>

</details>

  </td>
  <td><a href="https://github.com/alireza-abdii/AI-Comment-Filter/pull/1/files#diff-4d7c51b1efe9043e44439a949dfd92e5827321b34082903477fd04876edb7552">+14/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

**Comment by alireza-abdii** [comment]
add api feature for model
